test/unittest/rtps/history/ReaderHistoryTests.cpp

```diff
@@ -171,6 +176,64 @@ TEST_F(ReaderHistoryTests, get_change)
     }
 }
 
+TEST_F(ReaderHistoryTests, change_order)
+{
+    constexpr uint32_t n_writers = 4;
+    constexpr uint32_t changes_per_writer = 5;
+    std::vector<CacheChange_t*> changes;
+
+    // Create changes list
+    for (uint32_t i = 1; i <= n_writers; i++)
+    {
+        uint32_t ts = 0;
+        GUID_t writer_guid = GUID_t(GuidPrefix_t::unknown(), i);
+
+        for (uint32_t j = 1; j <= changes_per_writer; j++)
+        {
+            CacheChange_t* ch = new CacheChange_t(0);
+            ch->writerGUID = writer_guid;
+            ch->sequenceNumber = SequenceNumber_t(0, j);
+            ch->sourceTimestamp = rtps::Time_t(0, ts);
+
+            changes_list.push_back(ch);
```

Comment:
It should be `changes` instead of `changes_list`, shouldn't be?

Reply:
Test changed. See dc35cef